backend/app/interactem/app/operators/registry.py

```diff
@@ -0,0 +1,104 @@
+import urllib.parse
+from typing import Any
+
+from httpx import AsyncClient
+from jsonpath_ng import parse
+
+GITHUB_API_URL = "https://api.github.com"
+
+
+class ContainerRegistry:
+    def __init__(
+        self,
+        url: str,
+        username: str,
+        password: str,
+    ):
+        self._url = url
+        self._username = username
+        self._password = password
```

Comment:
is the password always a token (for other registries other than GH)? maybe we should call it `user_token` or something if so